docs/fork_choice.md

```diff
@@ -0,0 +1,79 @@
+2. [Save the attestation](https://eth2book.info/capella/part3/forkchoice/phase0/#update_latest_messages) as that validator's latest message. If there's one already, update the value.
+
+## Choosing forks
+
+We now have a store of each validator's latest vote, which allows LMD GHOST to work as a `get_head(store) -> Block` function.
+
+We first need to calculate each block's weight:
+
+- For leaf blocks, we calculate their weight by checking how much votes they have.
+- For each branch block we calculate its weight as the sum of the weight of every child, plus its own votes. We repeat this until we reach the root, which will be the last finalized block (there won't be any branches before, so there won't be any more fork-choice to perform).
+
+This way we calculate the weight not only for each block, but for the subtree were that block is the root.
+
+Afterwards, when we want to determine which is the head of the chain, we traverse the tree, starting from the root, and greedily (without looking further ahead) we go block by block chosing the sub-tree with the highest weight.
+
+Let's look at an example:
+
+```mermaid
+graph LR
+
+    Genesis --> A[A\nb=10\nw=50]
+    Genesis --> B[B\nb=20\nw=20]
+    A --> C[C\nb=15\nw=15]
+    A --> D[D\nb=25\nw=25]
+
+    classDef chosen fill: #666666
+    class Genesis chosen
+    class A chosen
+    class D chosen
+```
+
+Here, individual block weights are represented by "b", while subtree weights are represented by "w". Some observations:
+
+- $W = B$ for all leaf blocks, as leafs are their own whole subtree.
+- $W_A=W_C+W_B +B_A= B_B + B_C + B_A$
+- While the individual weight of $A$ is smaller than $B$, its children make the $A$ subtree heavier than the $B$ subtree, so its chosen by LMD GHOST over $B$.
+
+In general:
+
+$$W_N = B_N + \sum_i^{i \in \text{children}[N]}W_i$$
+
+## Slashing
+
+In the previous scheme, there are two rewards:
+
+- Proposer rewards, given to a proposer when their block is included in the chain. This also adds an incentive for them to try to predict the most-likely branch to be the canonical one.
+- Attester rewards, which are smaller. These are given if the blocks they attest to are included.
+
+These incentives, however, are not enough. To maximize their likelyhood of getting rewards, they may missbehave:
+
+- Proposers may propose a block for every current fork.
+- Attesters may attest to every current head in their local chains.
+
+These missbehaviors debilitate the protocol (they give weight to all forks) and no honest node running fork-choice would take part on them. To prevent them, nodes that are detected while doing them are slashed (punished), which means that they are excluded from the validator set and a portion of their stake is burned.
```

Comment:
```suggestion
These misbehaviors debilitate the protocol (they give weight to all forks) and no honest node running fork-choice would take part on them. To prevent them, nodes that are detected while doing them are slashed (punished), which means that they are excluded from the validator set and a portion of their stake is burned.
```